pulsar-functions/proto/src/main/proto/Function.proto

```diff
@@ -133,4 +133,5 @@ message Instance {
 message Assignment {
     Instance instance = 1;
     string workerId = 2;
+    bool removed = 3;
```

Comment:
Why do we take this approach. Instead we could just write an empty value to the key, which is the removal?

Reply:
Ya we should set the key of the assignment to be null so that during compaction the assignment will be removed.  If we don't do that, the assignment topic will still be littered with potentially many defunct assignments.

Reply:
> Why do we take this approach. Instead we could just write an empty value to the key, which is the removal?

yes, that's true. I didn't see that removal at compaction and I also had the concern of removing defunct assignments. I will fix it and will also fix compaction if necessary.